### PR TITLE
Reduce CI verbosity and complexity

### PR DESCRIPTION
**Pull request by saulshanabrook**
The goal with this PR is to make CI faster, easier to debug, and less flaky.

## TODO

- [x] Figure out how to get rendermine test to pass in CI when running with coverage. (removed cache loader)

## Results

* Time per build reduced from [~2 hours 20 min](https://travis-ci.org/jupyterlab/jupyterlab/builds/398427826) to [~1 hour 40 minutes](https://travis-ci.org/jupyterlab/jupyterlab/builds/398888284). This should decrease our load on Travis which will make our PRs build quicker.
* Lines printed in the JS tests decreased from [~6500](https://travis-ci.org/jupyterlab/jupyterlab/jobs/398427827) to [~600](https://travis-ci.org/jupyterlab/jupyterlab/jobs/398871685). This makes it easier to load the test output page and scroll through it, without it slowing down your browser.

## Changes
* Switched all tests to run with chrome headless and use [puppeteer](https://github.com/GoogleChrome/puppeteer) instead of firefox on selenium. This means we don't need to download firefox or any webdrivers in CI. 
* Upgrade karma
* Changed test scripts to use `set -ex; set -o pipefail`, instead of `set -x`.**
 * The `-e` will make sure the test exits with a failure code if any of the commands do.
 * The `-o pipefile` will make the test exit with a failure code if any piped command does.
* Added `-q` to pip install
* Disabled python logging for test server
* In CI, only show lerna warnings, not info messages about starting commands
* In CI hide all stdout of JS tests. This means, that test runs won't output anything, unless they fail. If they fail, lerna will output the failure to stderr, so it will still be seen in CI. See [this build](https://travis-ci.org/jupyterlab/jupyterlab/jobs/398852367) for an example of the failing output.
* ~~Revert karma timeouts to default. I am finding that longer timeouts/more retries just make it harder to debug.~~

## Future Improvements
There is still many ways to improve these metrics, but I think this is a good time to stop where we are at. Future work could include:

* Consolidating testing logic, including [moving services tests with rest of tests](https://travis-ci.org/jupyterlab/jupyterlab/jobs/397453412)
* Running all tests in one webpack build. This would hopefully speed up time spent type checking and building during all tests.
* Completely change how packages are organized so that it's a little less od-hoc how tests are built and dependencies are used. Maybe use webpack dll package to do this and investigate previous work done for extensions (https://github.com/jupyterlab/jupyterlab/issues/728). 
* Move all tests to CircleCI and do tests in stages building docker images, so that code only has to be built/type checking once, instead of again and again for each test type.

### CODE REVIEW

**Comment by saulshanabrook** [comment]
We could also auto cancel branch builds, so that if you push new changes to a branch the old ones builds get cancelled. We often are running up against travis's max concurrent builds, so this might help with that. 
screen shot 2018-06-15 at 11 51 23 am

**Comment by ian-r-rose** [comment]
:+1: on auto-canceling builds. I thought we were already doing that, in fact.

**Comment by saulshanabrook** [comment]
I flipped the switch.

File: scripts/travis_install.sh [hunk 1]
```diff
@@ -6,7 +6,6 @@ set -ev
 set -o pipefail
 
 # Building should work without yarn installed globally
-rm /home/travis/.yarn/bin/*
 !yarn --version
```
Comment (jasongrout) [inline scripts/travis_install.sh]: I think we need the `rm` above - it was specifically testing the case where yarn is not installed globally, fixing an issue we've had before.

File: scripts/travis_install.sh [hunk 2]
```diff
@@ -28,7 +27,7 @@ conda update -q conda
 conda info -a # for debugging
 
 conda remove --name test --all || true
-conda create -q -n test --quiet -c conda-forge notebook nodejs pytest python=$PYTHON
+conda create -q -n test --quiet -c conda-forge notebook pytest python=$PYTHON
```
Comment (jasongrout) [inline scripts/travis_install.sh]: Why use the system nodejs, just curious? For speed (since it's already installed)?

**Comment by jasongrout** [comment]
Sorry, I made some comments on some out-of-date commits, which no longer apply, so I deleted the comments.

**Comment by saulshanabrook** [comment]
@jasongrout The comments are still accurate on the newer commits.

Yeah I opted to use the system node to try to shave off some install time.

Ah, it's too bad we can't remove the `rm` command for yarn. I wonder if there is some creative way to make that command unavailable without requiring sudo. The [non sudo builds start up quicker](https://docs.travis-ci.com/user/reference/overview/#Virtualisation-Environment-vs-Operating-System).

It seems like I finally have chrome-headless working. Now I can run all the test commands in parallel, which speeds up things a ton on my laptop but might actually slow them down in CI, since it has less cores.

I am now trying to see if I can run all the tests in one command, like by running karma from the base `tests` directory to include all the files. Then it can do all the type checking at once, which might help things.

I also switched to pip instead of conda, in a hope that we would save time by not having to install miniconda, but I am not sure if this helps a ton, may have actually made things worse. 

I might try a CircleCI build as well, since it let's you use any Docker image as the base for your build, so that gives us more flexibility in terms of what's included.

**Comment by jasongrout** [comment]
> Ah, it's too bad we can't remove the rm command for yarn. I wonder if there is some creative way to make that command unavailable without requiring sudo. The non sudo builds start up quicker.

I suppose we could put something in the path to mask the system yarn that would throw an error. The point is that we want to make sure that the system yarn doesn't work, so we are testing that we are actually using our yarn, and work on a system without a working yarn.

**Comment by saulshanabrook** [comment]
I actually reverted to using the non container builds and re-enabled
deleting yarn. Container builds might run a bit slower anyway so I think
that's fine. Issue description is up to date, but my comment you quoted
isn't.

**Comment by saulshanabrook** [comment]
The `cache-loader` introduced in #4698 is causing me some problems when running with the chrome browser. I am getting failure with the rendermine tests when testing in the chrome browser and using the cache loader, some of the time. When I inspect the generated JS in the browser, it obviously has some problems. Here is an excerpt of `index.ts`:

```javascript
/**
 * A namespace for private data.
 */
var Private;
(function (Private) {
    Private.manager = new services_1.ServiceManager();
    Private.textFactory = new docregistry_1.TextModelFactory();
    Private.notebookFactory = new notebook_1.NotebookModelFactory({});
    var JSONRenderer = /** @class */ (function (_super) {
        __extends(JSONRenderer, _super);
        function JSONRenderer() {
            var _this = _super !== null && _super.apply(this, arguments) || this;
            _this.mimeType = 'text/html';
            return _this;
        }
        JSONRenderer.prototype.renderModel = function (model) {
            var source = model.data['application/json'];
            model.setData({ data: { 'text/html': json2html(source) } });
            return _super.prototype.renderModel.call(this, model);
        };
        return JSONRenderer;
    }(rendermime_1.RenderedHTML));
    var jsonRendererFactory = {
        mimeTypes: ['application/json'],
        safe: true,
        createRenderer: function (options) {
            debugger;
            return new JSONRenderer(options);
        }
    };
    Private.rendermime = new rendermime_1.RenderMimeRegistry({
        initialFactories: rendermime_1.standardRendererFactories
    });
    Private.rendermime.addFactory(jsonRendererFactory, 10);
})(Private || (Private = {}));
```

This file is messed up because `JSONRenderer` is transpiled to a function, but it still called with `new JSONRenderer`, like it is a class.  If we take away the `cache-loader`, it looks like this:

```
/**
 * A namespace for private data.
 */
var Private;
(function (Private) {
    Private.manager = new services_1.ServiceManager();
    Private.textFactory = new docregistry_1.TextModelFactory();
    Private.notebookFactory = new notebook_1.NotebookModelFactory({});
    class JSONRenderer extends rendermime_1.RenderedHTML {
        constructor() {
            super(...arguments);
            this.mimeType = 'text/html';
        }
        renderModel(model) {
            let source = model.data['application/json'];
            model.setData({ data: { 'text/html': json2html(source) } });
            return super.renderModel(model);
        }
    }
    const jsonRendererFactory = {
        mimeTypes: ['application/json'],
        safe: true,
        createRenderer(options) {
            debugger;
            return new JSONRenderer(options);
        }
    };
    Private.rendermime = new rendermime_1.RenderMimeRegistry({
        initialFactories: rendermime_1.standardRendererFactories
    });
    Private.rendermime.addFactory(jsonRendererFactory, 10);
})(Private || (Private = {}));
```

Which is more reasonable. 

The errors might be with with combing it with the thread-loader (https://github.com/webpack-contrib/cache-loader/issues/20).

I am going to remove it here and see if the tests still run in an acceptable amount of time.

**Comment by saulshanabrook** [comment]
@jasongrout This is ready for review again.

File: tests/test-notebook/src/model.spec.ts [hunk 3]
```diff
@@ -141,7 +141,7 @@ describe('@jupyterlab/notebook', () => {
           expect(called).to.be(true);
         });
 
-        it('should emit a `contentChanged` signal upon cell move', () => {
+        it.skip('should emit a `contentChanged` signal upon cell move', () => {
```
Comment (blink1073) [inline tests/test-notebook/src/model.spec.ts]: Intentional?

Comment (saulshanabrook) [reply tests/test-notebook/src/model.spec.ts]: Nope, not intentional. Good catch. Thank you for taking a look over this.

**Pull request merged**